cli/wallet/wallet.go

```diff
@@ -542,12 +555,45 @@ func importMultisig(ctx *cli.Context) error {
 		}
 	}
 
-	var label *string
 	if ctx.IsSet("name") {
 		l := ctx.String("name")
 		label = &l
 	}
-	acc, err := newAccountFromWIF(ctx.App.Writer, ctx.String("wif"), wall.Scrypt, label, pass)
+
+loop:
+	for _, pub := range pubs {
+		for _, wallAcc := range wall.Accounts {
+			if wallAcc.ScriptHash().Equals(pub.GetScriptHash()) {
+				if acc != nil {
+					// Multiple matching accounts found, fallback to WIF-based conversion.
+					acc = nil
+					break loop
+				}
+				acc = new(wallet.Account)
+				*acc = *wallAcc
+				accPub = pub
+			}
+		}
+	}
+
+	if acc != nil {
+		err = acc.ConvertMultisigEncrypted(accPub, m, pubs)
+		if err != nil {
+			return cli.NewExitError(err, 1)
+		}
+		if label != nil {
+			acc.Label = *label
+		}
+		if err := addAccountAndSave(wall, acc); err != nil {
+			return cli.NewExitError(err, 1)
+		}
+		return nil
+	}
+
+	if !ctx.IsSet("wif") {
+		return cli.NewExitError(errors.New("none of the provided public keys correspond to an existing key in the wallet, and no WIF is provided"), 1)
```

Comment:
We may adjust this error text to properly describe the behaviour. Something like “or multiple matching accounts found in the wallet” can be added before no-WIF warning.

Reply:
Added